src/shared/Core.Tests/EnvironmentTests.cs

```diff
@@ -123,5 +124,31 @@ public void PosixEnvironment_TryLocateExecutable_ExistsMultiple_ReturnTrueAndFir
             Assert.True(actualResult);
             Assert.Equal(expectedPath, actualPath);
         }
+
+        [PlatformFact(Platforms.MacOS)]
+        public void MacOSEnvironment_TryLocateExecutable_Paths_Are_Ignored()
+        {
+            string expectedPath = "/home/john.doe/bin/foo";
+            List<string> pathsToIgnore = new List<string>()
+            {
+                "/home/john.doe/bin/ignore/foo"
+            };
+
+            var fs = new TestFileSystem
+            {
+                Files = new Dictionary<string, byte[]>
+                {
+                    [pathsToIgnore.FirstOrDefault()] = Array.Empty<byte>(),
+                    [expectedPath] = Array.Empty<byte>(),
+                }
+            };
+            var envars = new Dictionary<string, string> {["PATH"] = PosixPathVar};
```

Comment:
Here `PosixPathVar` is `/home/john.doe/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin` which doesn't contain the `/home/john.doe/bin/ignore/` directory anyway, so this test will always return `true` since the existing code will not see the `ignore/` dir as existing.

Instead create two empty files at `/home/john.doe/bin/foo` and `/usr/local/bin/foo`, and then set `pathsToIgnore` as `/home/john.doe/bin/foo`.

We already have a test that verifies we should normally select the `john.doe` version ahead of the `usr/local` one, so this test will only pass if the ignored paths feature is correct.

Reply:
I _think_ I'm picking up what you're putting down. I was at least able to re-configure the test so that it was failing and returning the `/home/john.doe/bin/foo` before I passed `pathsToIgnore`. It then started passing after I passed `pathsToIgnore`. Let me know if further changes are needed, though.

Reply:
Looks good!